packages/hash/api/src/graphql/resolvers/knowledge/entity/entity.ts

```diff
@@ -183,6 +189,198 @@ export const getAllLatestEntitiesResolver: ResolverFn<
+    );
+  }
+
+  /** @todo - Encapsulate this transpilation in a shared function somewhere */
+  const filter: Filter = {
+    all: [
+      {
+        equal: [{ path: ["version"] }, { parameter: "latest" }],
+      },
+      {
+        equal: [{ path: ["archived"] }, { parameter: false }],
+      },
+    ],
+  };
+
+  if (operation.entityTypeId) {
+    filter.all.push({
+      equal: [
+        { path: ["type", "versionedUri"] },
+        { parameter: operation.entityTypeId },
+      ],
+    });
+  }
+
+  if (operation.multiFilter !== undefined && operation.multiFilter !== null) {
+    const subFilters: Filter[] = [];
+
+    /**
+     * @todo - support more than just "properties" as a field so we can filter on other things in the entity
+     *   https://app.asana.com/0/0/1203749694686236/f
+     */
+    for (const operationFilter of operation.multiFilter.filters) {
+      if (validateBaseUri(operationFilter.field).type !== "Ok") {
+        throw new ApolloError(
+          `Unexpected field entry in aggregation filter: ${operationFilter.field}`,
+        );
+      }
+
+      switch (operationFilter.operator) {
+        case "IS_EMPTY":
+          subFilters.push({
+            equal: [
+              { path: ["properties", operationFilter.field] },
+              /**
+               * @todo - This is incorrect and will return TRUE for properties with value `null`, not just missing ones
+               *   https://app.asana.com/0/0/1203749694686236/f
+               */
+              // @ts-expect-error -- Our typings don't allow us to pass `null` here, when they should.
+              null,
+            ],
+          });
+          break;
+        case "IS_NOT_EMPTY":
+          subFilters.push({
+            // @ts-expect-error -- Our typings don't allow us to pass `null` as a second param here, when they should.
+            not: [
+              {
+                equal: [
+                  { path: ["properties", operationFilter.field] },
+                  /**
+                   * @todo - This is incorrect and will return TRUE for properties with value `null`, not just missing ones
+                   *   https://app.asana.com/0/0/1203749694686236/f
+                   */
+                  null,
+                ],
+              },
+            ],
+          });
+          break;
+        case "IS":
+          /** @todo - The aggregation syntax should be updated to take any valid JSON Value */
+          // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- The BP aggregation syntax currently only accepts string values
+          if (operationFilter.value === null) {
+            throw new ApolloError(
+              "Equality checks with `null` are currently unsupported",
+            );
+          }
+          subFilters.push({
+            equal: [
+              /**
+               * @todo - This assumes this is just one field
+               *   https://app.asana.com/0/0/1203749694686236/f
+               */
```

Comment:
We need to update the BP aggregation type to take an array of fields (currently untracked), and update structural queries to support it (This is one of @TimDiekmann's next-ups (Asana Task (internal)))

cc @CiaranMn

Reply:
This has been rectified